src/types.rs

```diff
@@ -247,13 +247,27 @@ impl DateTime {
         minute: u8,
         second: u8,
     ) -> Result<DateTime, DateTimeRangeError> {
+        fn is_leap_year(year: u16) -> bool {
+            (year % 4 == 0) && ((year % 25 != 0) || (year % 16 == 0))
+        }
+
         if (1980..=2107).contains(&year)
             && (1..=12).contains(&month)
             && (1..=31).contains(&day)
             && hour <= 23
             && minute <= 59
-            && second <= 60
+            && second <= 58
```

Comment:
Clamp `second` into this range, so that we'll still accept input during leap seconds.